controllers/front/OrderOpcController.php

```diff
@@ -95,10 +95,21 @@ private function checkWhetherAllConditionsAreApproved()
         return true;
     }
 
+    public function getPaymentOptionsAction()
+    {
+        return $this->renderPaymentOptions();
+    }
+
     public function init()
     {
         parent::init();
 
+        if (($action = Tools::getValue('action'))) {
+            $result = $this->{$action . 'Action'}();
```

Comment:
Well if it doesn't exist it will cause a fatal error. I could check and then throw an exception instead but what's the point? If my action doesn't exist I need to implement it, I cannot have the code invent a default behaviour :)